### PR TITLE
Make all code pass tidy and critic checks

### PR DESCRIPTION
**Pull request by autarch**
This is mostly trivial changes, except for replacing hand-rolled eval-using tests with Test::Fatal.

### CODE REVIEW

File: lib/DateTime.pm [hunk 1]
```diff
@@ -1365,6 +1369,7 @@ sub jd { $_[0]->mjd + 2_400_000.5 }
         my $self    = shift;
         my $pattern = shift;
 
+        ## no critic (ontrolStructures::ProhibitCStyleForLoops)
```
Comment (autarch) [inline lib/DateTime.pm]: Weird, why did this work.

Comment (preaction) [inline lib/DateTime.pm]: IIRC if Perl::Critic doesn't see something it understands, it turns into `## no critic ()` which is just `## no critic`. I remember having the same problem.

File: lib/DateTime.pm [hunk 2]
```diff
@@ -1531,6 +1536,7 @@ sub subtract_datetime {
     );
 }
 
+## no critic (Subroutines::ProhibitManyArgs)
```
Comment (autarch) [inline lib/DateTime.pm]: No, all these `no critic` things needs to come before the code that critic complains about.

File: lib/DateTime.pm (hunk 1 above)
Comment (autarch) [inline lib/DateTime.pm]: Fixed in a rebase

File: lib/DateTime.pm (hunk 2 above)
Comment (preaction) [inline lib/DateTime.pm]: If I change line 1540 to `sub _adjust_for_positive_difference { ## no critic (Subroutines::ProhibitManyArgs)` and then remove line 1539 and run `tidyall lib/DateTime.pm`, it doesn't report any perlcritic problems (but the comment does get tidied on to the next line, which then also does not report any perlcritic problems). The perlcritic docs also say that `## no critic (...)` can come on the same line as is reported in the complaint.

Comment (autarch) [inline lib/DateTime.pm]: Ah, I see. I did it the way you suggested and pushed a rebase.

**Pull request merged**